### PR TITLE
update build to dune

### PR DESCRIPTION
**Pull request by rjbou**

### CODE REVIEW

**Comment by avsm** [comment]
Also, is it about time we just switched to the `master` branch for this, now that opam 1.x support is discontinued?

**Comment by rjbou** [comment]
lgtm to merge to master. @altgr?

File: opam-publish.opam [hunk 1]
```diff
@@ -11,16 +11,24 @@ homepage: "https://github.com/ocaml/opam-publish"
 bug-reports: "https://github.com/ocaml/opam-publish/issues"
 license: "LGPL-2.1 with OCaml linking exception"
 dev-repo: "git+https://github.com/ocaml/opam-publish.git#2.0"
-build: [ "jbuilder" "build" "-p" name ]
+build: [ "dune" "build" "-p" name "-j" jobs ]
 depends: [
+  "cmdliner" {build}
+  "dune" {build}
+  "lwt_ssl"
```
Comment (dra27) [inline opam-publish.opam]: Doesn't this want to be `{build}` as well, given that all the others are?

File: opam-publish.opam [hunk 2]
```diff
@@ -11,16 +11,24 @@ homepage: "https://github.com/ocaml/opam-publish"
...
+  "ocaml" {>= "4.02.3"}
```
Comment (dra27) [inline opam-publish.opam]: Unusually, perhaps this should be `{build}` although really what's meant is `{!ocaml:native | build}` which I don't think is permitted.

File: opam-publish.opam [hunk 3]
```diff
@@ -11,16 +11,24 @@ homepage: "https://github.com/ocaml/opam-publish"
...
   "opam-core" {build & >= "2.0.0"}
   "opam-format" {build & >= "2.0.0"}
   "opam-state" {build & >= "2.0.0"}
-  "ocamlfind" {build}
-  "cmdliner" {build}
-  "lwt_ssl"
-  ("ssl" {build & = "0.5.5"} | "tls")
   ("github" {build & >= "2.0.0" & < "3.0.0"} |
    "github-unix" {build & >= "3.0.0"})
+  ("ssl" {build & = "0.5.5"} | "tls")
```
Comment (dra27) [inline opam-publish.opam]: `{build}` after `tls` too

File: opam-publish.opam [hunk 4]
```diff
@@ -11,16 +11,24 @@ homepage: "https://github.com/ocaml/opam-publish"
...
 ]
-tags: [ "flags:plugin" ]
+flags: plugin
+synopsis: " A tool to ease contributions to opam repositories"
```
Comment (dra27) [inline opam-publish.opam]:
```suggestion
synopsis: "A tool to ease contributions to opam repositories"
```

File: opam-publish.opam [hunk 5]
```diff
@@ -11,16 +11,24 @@ homepage: "https://github.com/ocaml/opam-publish"
...
+description:
+"""
+opam-publish is a helper to publish your package: it checks that your package
+is well filled (lint pass), checks and adds archive url and its associated
+checksum, fills a pull request and submits it.
+"""
```
Comment (dra27) [inline opam-publish.opam]: FWIW:

```
description:
"""
opam-publish automates publishing packages to package repositories: it checks that the
opam file is complete using `opam lint`, verifies and adds the archive URL and its
checksum and files a GitHub pull request for merging it.
"""
```

**Comment by rjbou** [comment]
@dra thx!

**Pull request merged**